src/plugins/condition/StyleRuleManager.js

```diff
@@ -74,26 +85,33 @@ export default class StyleRuleManager extends EventEmitter {
             if (foundStyle !== this.currentStyle) {
                 this.currentStyle = foundStyle;
             }
+            this.updateDomainObjectStyle();
         } else {
-            if (this.currentStyle !== this.defaultStyle) {
-                this.currentStyle = this.defaultStyle;
-            }
+            this.applyStaticStyle();
         }
-
-        this.updateDomainObjectStyle();
     }
 
     updateDomainObjectStyle() {
-        this.emit('conditionalStyleUpdated', this.currentStyle)
+        if (this.callback) {
+            this.callback(Object.assign({}, this.currentStyle));
```

Comment:
curious if we need to pass in a callback function to this class instead of using this event (or a different one if needed) that is emitted

Reply:
Unsure what you mean? We are passing a callback function to this class.